src/visitable.cpp

```diff
@@ -164,7 +148,8 @@ VisitResponse visitable<vehicle_cursor>::visit_items( const std::function<VisitR
 }
 
 template <>
-VisitResponse visitable<vehicle_selector>::visit_items( const std::function<VisitResponse( item *, item *, const tripoint* )>& func )
+VisitResponse visitable<vehicle_selector>::visit_items(
```

Comment:
Regarding the parent parameter: to remove an item, you have to implement a separate function anyway to handle top-level items (not inside another container but directly on the map/in the cargo part).

> The third argument is currently unused

It's incomplete information anyway. Items on the map, in a cargo part and on a character may all have the same position, how are you going to distinguish that. The code that calls `visit_items` should know where the item is and can always forward that knowledge to the functor. And it can forward exactly the required knowledge and nothing more.